pkg/controller/apicast/apicast_logic_reconciler.go

```diff
@@ -0,0 +1,431 @@
+	// in OwnerRef?
+	err = r.Client().Update(context.TODO(), &adminPortalCredentialsSecret)
+	if err != nil {
+		return apicast.AdminPortalCredentials{}, err
+	}
+
+	result := apicast.AdminPortalCredentials{
+		URL:         adminPortalURL,
+		AccessToken: adminPortalAccessToken,
+	}
+	return result, nil
+}
+
+func (r *APIcastLogicReconciler) internalAPIcast() (apicast.APIcast, error) {
+	apicastFullName := "apicast-" + r.APIcastCR.Name
+	apicastExposedHost := apicast.ExposedHost{}
+	if r.APIcastCR.Spec.ExposedHost != nil {
+		apicastExposedHost.Host = r.APIcastCR.Spec.ExposedHost.Host
+		apicastExposedHost.TLS = r.APIcastCR.Spec.ExposedHost.TLS
+	}
+	apicastOwnerRef := asOwner(r.APIcastCR)
+
+	adminPortalCredentials, err := r.reconcileAdminPortalCredentials()
+	if err != nil {
+		return apicast.APIcast{}, err
+	}
+
+	var deploymentEnvironment *string
+	if r.APIcastCR.Spec.DeploymentEnvironment != nil {
+		res := string(*r.APIcastCR.Spec.DeploymentEnvironment)
+		deploymentEnvironment = &res
+	}
+
+	var logLevel *string
+	if r.APIcastCR.Spec.LogLevel != nil {
+		res := string(*r.APIcastCR.Spec.LogLevel)
+		logLevel = &res
+	}
+
+	var managementAPIScope *string
+	if r.APIcastCR.Spec.ManagementAPIScope != nil {
+		res := string(*r.APIcastCR.Spec.ManagementAPIScope)
+		managementAPIScope = &res
+	}
+	apicastResult := apicast.APIcast{
+		DeploymentName:                 apicastFullName,
+		ServiceName:                    apicastFullName,
+		Replicas:                       int32(*r.APIcastCR.Spec.Replicas),
+		AppLabel:                       "apicast",
+		ServiceAccountName:             *r.APIcastCR.Spec.ServiceAccount,
+		Image:                          *r.APIcastCR.Spec.Image,
+		ExposedHost:                    apicastExposedHost,
+		Namespace:                      r.APIcastCR.Namespace,
+		OwnerReference:                 &apicastOwnerRef,
+		AdminPortalCredentials:         adminPortalCredentials,
+		DeploymentEnvironment:          deploymentEnvironment,
+		DNSResolverAddress:             r.APIcastCR.Spec.DNSResolverAddress,
+		EnabledServices:                r.APIcastCR.Spec.EnabledServices,
+		ConfigurationLoadMode:          r.APIcastCR.Spec.ConfigurationLoadMode,
+		LogLevel:                       logLevel,
+		PathRoutingEnabled:             r.APIcastCR.Spec.PathRoutingEnabled,
+		ResponseCodesIncluded:          r.APIcastCR.Spec.ResponseCodesIncluded,
+		CacheConfigurationSeconds:      r.APIcastCR.Spec.CacheConfigurationSeconds,
+		ManagementAPIScope:             managementAPIScope,
+		OpenSSLPeerVerificationEnabled: r.APIcastCR.Spec.OpenSSLPeerVerificationEnabled,
+	}
+
+	return apicastResult, err
+}
+
+func (r *APIcastLogicReconciler) namespacedName(object metav1.Object) types.NamespacedName {
+	return types.NamespacedName{
+		Name:      object.GetName(),
+		Namespace: object.GetNamespace(),
+	}
+}
+
+func (r *APIcastLogicReconciler) getAPIcast(request reconcile.Request) (*appsv1alpha1.APIcast, error) {
```

Comment:
used?

Reply:
nope. It was moved to the controller itself. I'll remove it :+1: